src/one/nio/ws/extension/ExtensionRequestParser.java

```diff
@@ -0,0 +1,84 @@
+package one.nio.ws.extension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  The relevant ABNF for the Sec-WebSocket-Extensions is as follows:
+ *      extension-list = 1#extension
+ *      extension = extension-token *( ";" extension-param )
+ *      extension-token = registered-token
+ *      registered-token = token
+ *      extension-param = token [ "=" (token | quoted-string) ]
+ *          ; When using the quoted-string syntax variant, the value
+ *          ; after quoted-string unescaping MUST conform to the
+ *          ; 'token' ABNF.
+ *
+ *  The limiting of parameter values to tokens or "quoted tokens" makes
+ *  the parsing of the header significantly simpler and allows a number
+ *  of short-cuts to be taken.
+ *
+ * @author <a href="mailto:vadim.yelisseyev@gmail.com">Vadim Yelisseyev</a>
+ */
+public class ExtensionRequestParser {
+
+    public static List<ExtensionRequest> parse(String header) {
```

Comment:
It seems unit test for this parsing would be valuable.

Reply:
added tests for this parser